File: Array_String/1-unique/Python/unique_v2.py
def isUniqueChars(string):
    checker = 0
    for char in string:
        val = ord(char) - ord('a')
        if checker & (1 << val) > 0:
            return False # already found this character in the string
        checker |= (1 << val)
    return True

if __name__ == '__main__':    
    input_string = input("Enter a string: ")
    if isUniqueChars(input_string):
        print("All characters in the string are unique.")
    else:
        print("There are duplicate characters in the string.")

File: Array_String/1-unique/Cpp/unique_v2.cpp
#include <iostream>
#include <string>
using namespace std;

bool isUniqueChars(string str) {
    int checker = 0;

    for (int i = 0; i < str.length(); i++) {
        int val = str[i] - 'a';

        if ((checker & (1 << val)) > 0) return false; // already found this character in the string

        checker |= (1 << val);
    }
    return true;
}

int main() {
    string inputString;
    cout << "Enter a string: ";
    cin >> inputString;

    if (isUniqueChars(inputString)) {
        cout << "All characters in the string are unique." << endl;
    }
    else {
        cout << "There are duplicate characters in the string." << endl;
    }
    return 0;
}
